typescript/src/chain.ts

```diff
@@ -45,6 +46,20 @@ export interface Bridge {
     deposit: Deposit
   ): Promise<void>
 
+  /**
+   * Submits a redemption transaction proof to the on-chain contract.
+   * @param redemptionTx - Redemption transaction data
+   * @param redemptionProof - Redemption proof data
+   * @param mainUtxo - Data of the wallets main UTXO
+   * @param walletPubKeyHash - 20-byte public key hash of the wallet
+   */
+  submitRedemptionProof(
+    redemptionTx: DecomposedRawTransaction,
+    redemptionProof: Proof,
+    mainUtxo: UnspentTransactionOutput,
+    walletPubKeyHash: BytesLike
```

Comment:
Just like I mentioned in `index.ts`, let's use plain-text `walletPublicKey`. This will be consistent with what we have in `revealDeposit` where `deposit: Deposit` has a plain-text key of the wallet too. Bridge implementation will do the appropriate hashing under the hood. Also, tying the API with plain-text key of the wallet seems more flexible since we'll also need to deal with frauds in future where the key is needed for signature verification.

Reply:
Done in 3491b959b2105173486d028e91926e68abfeaa6b.